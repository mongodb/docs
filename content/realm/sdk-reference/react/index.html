<!doctype html>
<head>
  <meta http-equiv="refresh" content="0; url=https://mongodb.com/docs/realm-sdks/react/latest" />
</head>
<body>
  <p><a href="https://mongodb.com/docs/realm-sdks/react/latest">Redirect</a></p>
</body>
