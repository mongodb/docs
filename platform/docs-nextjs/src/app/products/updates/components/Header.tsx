'use client';
import { css, cx } from '@leafygreen-ui/emotion';
import { H2 } from '@leafygreen-ui/typography';
import { Body } from '@leafygreen-ui/typography';
// import Button from '@leafygreen-ui/button';
import { palette } from '@leafygreen-ui/palette';
import HeaderBackground from './HeaderBackground';
import { theme } from '@/styles/theme';

const headerStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
  height: 283px;
  width: 100%;
  position: relative;

  @media ${theme.screenSize.mediumAndUp} {
    height: 230px;
  }

  @media ${theme.screenSize.largeAndUp} {
    height: 384px;
  }
`;

const headerContentStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 782px;
  z-index: 2;
`;

const titleStyle = css`
  font-size: 32px;
  line-height: 40px;
  color: ${palette.black};
  margin-bottom: 8px;
  max-width: 326px;
  text-align: center;

  @media ${theme.screenSize.largeAndUp} {
    font-size: 64px;
    line-height: 72px;
    max-width: 763px;
  }
`;

const bodyStyle = css`
  text-align: center;
  margin-bottom: 20px;
  font-size: 13px;
  line-height: 20px;
  color: ${palette.black};
  max-width: 326px;

  @media ${theme.screenSize.mediumAndUp} {
    max-width: 520px;
  }

  @media ${theme.screenSize.largeAndUp} {
    font-size: 16px;
    line-height: 28px;
    max-width: 763px;
  }
`;

// const buttonStyle = css`
//     font-size: 13px;
//     line-height: 20px;
//     padding: 4px 12px;

//     @media ${theme.screenSize.mediumAndUp} {
//         padding: 8px 12px;
//     }

//     @media ${theme.screenSize.largeAndUp} {
//         font-size: 18px;
//         line-height: 24px;
//         padding: 12px 16px;
//         height: 48px;
//     }
// `;

const Header = () => {
  return (
    <div className={cx(headerStyle)}>
      <HeaderBackground />
      <div className={cx(headerContentStyle)}>
        <H2 className={cx(titleStyle)}>What&apos;s New at MongoDB?</H2>
        <Body className={cx(bodyStyle)}>
          Check out the latest updates in MongoDB â€“ including improvements to the developer experience, expanded
          workload support, app modernization tools, and more.
        </Body>
        {/* <Button variant="baseGreen" className={cx(buttonStyle)}>Subscribe via RSS Feed</Button> */}
      </div>
    </div>
  );
};

export default Header;
