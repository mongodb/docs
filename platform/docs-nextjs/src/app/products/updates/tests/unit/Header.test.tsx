import { render } from '@testing-library/react';
import Header from '../../components/Header';

describe('Header component', () => {
  it('renders correctly', () => {
    const wrapper = render(<Header />);
    expect(wrapper.container.querySelector('h2')).toHaveTextContent("What's New at MongoDB");
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
