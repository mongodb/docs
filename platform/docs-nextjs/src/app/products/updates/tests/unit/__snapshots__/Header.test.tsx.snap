// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header component renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
  height: 283px;
  width: 100%;
  position: relative;
}

@media not all and (max-width: 767px) {
  .emotion-0 {
    height: 230px;
  }
}

@media not all and (max-width: 1024px) {
  .emotion-0 {
    height: 384px;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  background-color: #E3FCF7;
  border-radius: 0 0 40px 40px;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}

@media not all and (max-width: 767px) {
  .emotion-1 {
    height: 230px;
    border-radius: 0 0 80px 80px;
  }
}

@media not all and (max-width: 1024px) {
  .emotion-1 {
    height: 384px;
    border-radius: 0 0 80px 80px;
  }
}

.emotion-2 {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
}

@media not all and (max-width: 767px) {
  .emotion-2 {
    top: 10px;
  }
}

@media not all and (max-width: 1440px) {
  .emotion-2 {
    top: 0;
  }
}

.emotion-3 {
  display: block;
}

@media not all and (max-width: 767px) {
  .emotion-3 {
    display: none;
  }
}

.emotion-4 {
  display: none;
}

@media not all and (max-width: 767px) {
  .emotion-4 {
    display: block;
  }
}

@media not all and (max-width: 1440px) {
  .emotion-4 {
    display: none;
  }
}

.emotion-5 {
  display: none;
}

@media not all and (max-width: 1440px) {
  .emotion-5 {
    display: block;
  }
}

.emotion-6 {
  position: absolute;
  top: 0px;
  right: 0px;
  z-index: 1;
}

@media not all and (max-width: 767px) {
  .emotion-6 {
    top: 0px;
    right: -10px;
  }
}

@media not all and (max-width: 1440px) {
  .emotion-6 {
    top: -20px;
    right: -20px;
  }
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 782px;
  z-index: 2;
}

.emotion-11 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 32px;
  line-height: 40px;
  font-weight: 400;
  font-family: 'MongoDB Value Serif','Times New Roman',serif;
  color: #00684A;
  font-size: 32px;
  line-height: 40px;
  color: #001E2B;
  margin-bottom: 8px;
  max-width: 326px;
  text-align: center;
}

@media not all and (max-width: 1024px) {
  .emotion-11 {
    font-size: 64px;
    line-height: 72px;
    max-width: 763px;
  }
}

.emotion-12 {
  margin: unset;
  font-family: 'Euclid Circular A','Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #001E2B;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  font-weight: 400;
  text-align: center;
  margin-bottom: 20px;
  font-size: 13px;
  line-height: 20px;
  color: #001E2B;
  max-width: 326px;
}

.emotion-12 strong,
.emotion-12 b {
  font-weight: 600;
}

@media not all and (max-width: 767px) {
  .emotion-12 {
    max-width: 520px;
  }
}

@media not all and (max-width: 1024px) {
  .emotion-12 {
    font-size: 16px;
    line-height: 28px;
    max-width: 763px;
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
      data-testid="header-background-img"
    >
      <div
        class="emotion-2"
      >
        <img
          alt="header-left"
          class="emotion-3"
          data-nimg="1"
          decoding="async"
          height="220"
          loading="lazy"
          src="/products/updates/header-mobile-left.svg"
          style="color: transparent;"
          width="220"
        />
        <img
          alt="header-left"
          class="emotion-4"
          data-nimg="1"
          decoding="async"
          height="320"
          loading="lazy"
          src="/products/updates/header-tablet-left.svg"
          style="color: transparent;"
          width="320"
        />
        <img
          alt="header-left"
          class="emotion-5"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/products/updates/header-desktop-left.svg"
          style="color: transparent;"
          width="300"
        />
      </div>
      <div
        class="emotion-6"
      >
        <img
          alt="header-right"
          class="emotion-3"
          data-nimg="1"
          decoding="async"
          height="160"
          loading="lazy"
          src="/products/updates/header-mobile-right.svg"
          style="color: transparent;"
          width="160"
        />
        <img
          alt="header-right"
          class="emotion-4"
          data-nimg="1"
          decoding="async"
          height="200"
          loading="lazy"
          src="/products/updates/header-tablet-right.svg"
          style="color: transparent;"
          width="200"
        />
        <img
          alt="header-right"
          class="emotion-5"
          data-nimg="1"
          decoding="async"
          height="360"
          loading="lazy"
          src="/products/updates/header-desktop-right.svg"
          style="color: transparent;"
          width="360"
        />
      </div>
    </div>
    <div
      class="emotion-10"
    >
      <h2
        class="emotion-11"
      >
        What's New at MongoDB?
      </h2>
      <p
        class="emotion-12"
      >
        Check out the latest updates in MongoDB â€“ including improvements to the developer experience, expanded workload support, app modernization tools, and more.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
